keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1525 @@
+utilize the idle resources of the other containers within a pod. 
+
+Since resource bursts across containers are not synchronized, the aggregate
+resource usage of the pod, in this case, is generally lower than the sum of the
+individual container's peak usage. By setting the pod level limit to a percentile
+of this aggregate usage, we can achieve better resource utilization and avoid
+unnecessary over-allocation.
+
+### Notes/Constraints/Caveats
+
+1. cgroupv1 has been moved to maintenance mode since Kubernetes version 1.31.
+   Hence this feature will be supported only on cgroupv2.
+2. Pod Level Resource Specifications is an **opt-in** feature, and will have no
+effect on existing deployments. Only deployments that explicitly require this
+functionality should turn it on by setting the relevant resource section at
+pod-level in the Pod specification.
+
+### Risks and Mitigations
+
+1. Since Pod Level Resource Specifications is an **opt-in** feature, merging the
+   feature related changes won't impact existing workloads. Moreover, the feature
+   will be rolled out gradually, beginning with an alpha release for testing and
+   gathering feedback. This will be followed by beta and GA releases as the
+   feature matures and potential problems and improvements are addressed.
+
+2. While this feature doesn't alter the existing cgroups structure, it does
+   change how pod-level cgroup values are determined. Currently, Kubernetes
+   derived these values from the container-level cgroup settings. However, with
+   Pod Level Resource Specifications enabled, pod-level cgroup settings will be
+   directly set based on the values specified in the pod's resource spec stanza.
+
+   This means that any workloads or tools that depend on reading or interpreting
+   container cgroup values might observe different behavior if they start using
+   pod level resource spec without specifying container level settings. This
+   aspect of the feature will be clearly highlighted in the feature documentation
+   to make users aware of this change and the potential impact on their workloads.
+   This way, they can adjust the pod level and container level settings
+   accordingly as needed.
+
+## Design Details
+
+### Components/Features changes
+
+#### Cgroup Structure Remains unchanged
+
+This KEP introduces pod-level resource specifications but does not alter the
+fundamental cgroup structure. Currently, pod level cgroups are derived from the
+resource requests and limits defined at the container level in the PodSpec. With
+this KEP, the PodSpec is extended to include pod-level resource requests and
+limits. These values will be directly used to configure the pod-level cgroup,
+providing a more direct and explicit way to manage the overall resource
+consumption of the pod.
+
+#### PodSpec API changes
+
+New field in `PodSpec`
+
+```
+type PodSpec struct {
+...
+  // Compute resource requirements.
```

Comment:
the field description will need to be expanded to describe the feature gate needed, the lack of cgroupv1 support, the limitation that only cpu and memory are currently supported and a relationship with other resource definitions (maybe in a form of a link)